inference-test-tool/test_inference_mask.py

```diff
@@ -84,17 +86,18 @@ def generate_images_with_masks(dicom_images, inference_results, response_json, o
     create_folder(output_folder)
 
     offset = 0
-    last_timepoint = None
     for index, image in enumerate(images):
         dcm = pydicom.dcmread(image.path)
         pixels = get_pixels(dcm)
 
-        if image.timepoint is not None and last_timepoint != image.timepoint:            
-            # Reset offset when 
-            offset = 0
-            last_timepoint = image.timepoint
+        # Reshape and add alpha
+        pixels = np.reshape(pixels, (-1, 3))
+        pixels = np.hstack((pixels, np.reshape(np.full(pixels.shape[0], 255, dtype=np.uint8), (-1, 1))))
 
-        for mask_index, (mask, json_part) in enumerate(zip(masks, response_json["parts"])):            
+        for mask_index, (mask, json_part) in enumerate(zip(masks, response_json["parts"])):
+            # If the input holds multiple timepoints but the result only includes 1 timepoint
+            if image.timepoint is not None and image.timepoint > 0 and json_part['binary_data_shape']['timepoints'] == 1:
```

Comment:
For cases where the output is a single timepoint but the input includes multiple timepoints, we show the output mask only on the first timepoint of the input.

The other supported case is when the input and output have the same shape